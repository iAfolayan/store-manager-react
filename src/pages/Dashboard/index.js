import React, { Component, Fragment } from 'react';
import Footer from '../../components/Footer';
import Navigation from '../../components/Navbar';
import Filter from '../../helpers/Filter';
import './Dashboard.scss';
import Products from '../../components/Products';
import Cart from '../../components/Products/cart';

export class Dashboard extends Component {
  state = {
    filterQuery: ''
  };

  handleChange = ({ target }) => {
    this.setState({
      filterQuery: target.value
    });
  }

  render() {
    const { filterQuery } = this.state;
    return (
      <Fragment>
      <div className="form-group row">
        <div className="col-md-2">
          <Navigation />
        </div>
        <div className="col-md-10">
          <Filter handleChange={this.handleChange} value={filterQuery} />
          <Cart />
          <Products filterQuery={filterQuery} />
          <Footer footerClassName="footer inner" />
        </div>
      </div>
      </Fragment>
    );
  }
}

export default Dashboard;
