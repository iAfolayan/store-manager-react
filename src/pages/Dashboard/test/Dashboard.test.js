import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from '../index';
import sinon from 'sinon';

const event = {
  target: {
      name:"ghjk",
  },
  getState: () => {}
}

describe('Dashboard', () => {
  test('should rendered Dashboard without crashing', () => {
    const wrapper = shallow(
      <Dashboard />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it.only('should simulate on change event', () => {
      const wrapper = shallow(<Dashboard />)
      sinon.spy(wrapper.instance(), 'handleChange');
      wrapper.instance().handleChange(event);
      expect(wrapper.instance().handleChange.calledOnce)
        .toEqual(true);
      expect(wrapper.instance().handleChange.calledWith(event));
    });
});