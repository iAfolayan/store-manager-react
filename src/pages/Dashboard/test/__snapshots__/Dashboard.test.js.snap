// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should rendered Dashboard without crashed 1`] = `
<Fragment>
  <div
    className="form-group row"
  >
    <div
      className="col-md-2"
    >
      <Connect(Navigation) />
    </div>
    <div
      className="col-md-10"
    >
      <Filter />
      <Connect(Products) />
      <Footer
        footerClassName="footer inner"
      />
    </div>
  </div>
</Fragment>
`;
